scipy/stats/_multivariate.py

```diff
@@ -1141,9 +1141,11 @@ def rvs(self, mean=None, rowcov=1, colcov=1, size=1, random_state=None):
         rowchol = scipy.linalg.cholesky(rowcov, lower=True)
         colchol = scipy.linalg.cholesky(colcov, lower=True)
         random_state = self._get_random_state(random_state)
-        std_norm = random_state.standard_normal(size=(dims[1], size, dims[0]))
-        roll_rvs = np.tensordot(colchol, np.dot(std_norm, rowchol.T), 1)
-        out = np.rollaxis(roll_rvs.T, axis=1, start=0) + mean[np.newaxis, :, :]
+        std_norm = random_state.standard_normal(size=(size, dims[0], dims[1]))
+        out = np.einsum('jp,ipq,kq->ijk', 
+                        rowchol, std_norm, colchol,
+                        optimize=True)
+        out += mean[np.newaxis, :, :]
```

Comment:
You should not need the `np.newaxis` at the beginning, it should be taken care of by broadcasting.

It might also be worth checking if `np.matmul` would be fast, but in theory the `optimize=True` should do better because by using `matmul` you'd have to choose the order ahead of time, unless you know for example that something is square so that the order doesn't matter.

Reply:
It turns out that because of the nature of the problem (multiply `n`x`n` times `size`x`n`x`p` times `p`x`p`), `rowchol @ (std_norm @ colchol.T)`, `(rowchol @ std_norm) @ colchol.T`, and `np.einsum('jp,ipq,kq->ijk',  rowchol, std_norm, colchol, optimize=True)`, all require the same number of FLOPS (`2 * size * n * p (n + p)`).